pkg/controller/proxyconfig/validation.go

```diff
@@ -48,5 +52,63 @@ func (r *ReconcileProxyConfig) ValidateProxyConfig(proxyConfig *configv1.ProxySp
 		}
 	}
 
+	if isSpecReadinessEndpoints(proxyConfig) {
+		for _, endpoint := range proxyConfig.ReadinessEndpoints {
+			scheme, err := validation.URI(endpoint)
+			if err != nil {
+				return fmt.Errorf("invalid URI for endpoint %s: %v", endpoint, err)
+			}
+			switch {
+			case scheme == proxyHTTPScheme:
+				// TODO: Add case for proxyHTTPSScheme when CA support is merged.
+				if err := validateHTTPReadinessEndpoint(endpoint); err != nil {
+					return fmt.Errorf("readinessEndpoint probe failed for endpoint %s", endpoint)
+				}
+			default:
+				// TODO: Update error to include proxyHTTPSScheme when CA support is merged.
+				return fmt.Errorf("readiness endpoints requires a %q URI sheme", proxyHTTPScheme)
+			}
+		}
+	}
+
+	return nil
+}
+
+// validateHTTPReadinessEndpoint validates an http readinessEndpoint endpoint.
+func validateHTTPReadinessEndpoint(endpoint string) error {
+	if err := validateHTTPReadinessEndpointWithRetries(endpoint, proxyProbeMaxRetries); err != nil {
+		return err
+	}
+
 	return nil
 }
+
+// validateHTTPReadinessEndpointWithRetries tries to validate an http
+// endpoint in a finite loop and returns the last result if it never succeeds.
+func validateHTTPReadinessEndpointWithRetries(endpoint string, retries int) error {
+	var err error
+	for i := 0; i < retries; i++ {
+		err = runHTTPReadinessProbe(endpoint)
+		if err == nil {
+			return nil
+		}
+	}
+
+	return err
+}
+
+// runHTTPReadinessProbe issues an http GET request to endpoint and returns
+// an error if a 2XX or 3XX http status code is not returned.
+func runHTTPReadinessProbe(endpoint string) error {
+	resp, err := http.Get(endpoint)
```

Comment:
how do we know this get is actually going through the spec proxy value that's being configured?  i'm pretty sure it's not.

Reply:
Thanks for catching this. Yes, I need to have CNO request proxy injection. Let me submit a PR for that.

Reply:
@bparees https://github.com/openshift/cluster-network-operator/pull/277/files#diff-1502a5b3db6bd911880e1d0d9346b561R107-R142 added proxy support to the http readiness probe.

Reply:
CNO does not need proxy injection. Instead the http probe transport has been updated to use a proxy from `proxy.spec.httpProxy`.